### PR TITLE
Add support for json export

### PR DESCRIPTION
**Pull request by yves-chan**
Add support for json config exporting

### CODE REVIEW

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/EncassWriter.java [hunk 1]
```diff
@@ -33,21 +30,12 @@
 
     @Override
     public void write(Bundle bundle, File rootFolder) {
-        File configFolder = new File(rootFolder, "config");
-        documentFileUtils.createFolder(configFolder.toPath());
+        Map<String, Encass> encassBeans = bundle.getEntities(EncassEntity.class)
+                .values()
+                .stream()
+                .collect(Collectors.toMap(EncassEntity::getPath, this::getEncassBean));
```
Comment (joao-borges) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/EncassWriter.java]: jsonTools is passed as argument, so you can pick the extension inside `writeFile` method 
Doing this we don't need to remember to concatenate the extension before calling the method in all entities, let the writer decide what needs to be.

Comment (yves-chan) [reply src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/EncassWriter.java]: Good point, I'll make this adjustment

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 2]
```diff
@@ -54,4 +62,24 @@ public void serialize(Object value, JsonGenerator gen, SerializerProvider serial
...
+
+    public void setOutputType(String outputType) {
+        if (JSON.equalsIgnoreCase(outputType)) {
+            this.outputType = JSON;
+            fileExtension = JSON_FILE_EXTENSION;
+        } else if (YAML.equalsIgnoreCase(outputType)) {
+            this.outputType = YAML;
+            fileExtension = YAML_FILE_EXTENSION;
+        } else {
+            //Junit bug where if the previous test sets outputType to JSON,
```
Comment (yves-chan) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: In the unit tests, I'm not sure how to reset the JsonTools instance, so for now in order to pass, the tests have to be run in a specific order :/. Any ideas?

Comment (joao-borges) [reply src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: the class CAGatewayExportTest is testing the plugin execution, you may want to create a separate test class only to test the ExplodeBundleTask class, then you can expose the constructor, and instantiate it with other defaults, but using a custom instance of JsonTools.

Comment (yves-chan) [reply src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Thanks I will take a look into creating a explodebundletask class

**Pull request merged**